lib/ex_webrtc/rtp/jitter_buffer.ex

```diff
@@ -0,0 +1,166 @@
+defmodule ExWebRTC.RTP.JitterBuffer do
+  @moduledoc """
+  Buffers and reorders RTP packets based on `sequence_number`, introducing controlled latency
+  in order to combat network jitter and improve the QoS.
+  """
+
+  # Heavily inspired by:
+  # https://github.com/membraneframework/membrane_rtp_plugin/blob/23f3279540aea7dea3a194fd5a1680c2549aebae/lib/membrane/rtp/jitter_buffer.ex
+
+  alias ExWebRTC.RTP.JitterBuffer.PacketStore
+  alias ExRTP.Packet
+
+  @default_latency_ms 200
+
+  @typedoc """
+  Options that can be passed to `new/1`.
+
+  * `latency` - latency introduced by the buffer, in milliseconds. `#{@default_latency_ms}` by default.
+  """
+  @type options :: [latency: non_neg_integer()]
+
+  @typedoc """
+  Time (in milliseconds) after which `handle_timer/1` should be called.
+  Can be `nil`, in which case no timer needs to be set.
+  """
+  @type timer :: non_neg_integer() | nil
+
+  @typedoc """
+  The 3-element tuple returned by all functions other than `new/1`.
+
+  * `buffer` - `t:#{inspect(__MODULE__)}.t/0`.
+  * `packets` - a list with packets flushed from the buffer as a result of the function call. May be empty.
+  * `timer_duration_ms` - see `t:timer/0`.
+
+  Generally speaking, all results of this type can be handled in the same way.
+  """
+  @type result :: {buffer :: t(), packets :: [Packet.t()], timer_duration_ms :: timer()}
```

Comment:
Generally, the structure you're modifying is returned as the last thing (like in `Map.pop` for instance. We should probably stick to this convention